lib/node.js

```diff
@@ -299,6 +310,90 @@ class Node {
     this.edgesIn.forEach(edge => edge.reload(true))
   }
 
+  // node.explain(nodes seen already, edge we're trying to satisfy
+  // if edge is not specified, it lists every edge into the node.
+  explain (edge = null, seen = []) {
+    if (this[_explanation])
+      return this[_explanation]
+
+    return this[_explanation] = this[_explain](edge, seen)
+  }
+
+  [_explain] (edge, seen) {
+    if (this.isRoot && !this.sourceReference) {
+      return {
+        location: this.path
+      }
+    }
+
+    const why = {
+      name: this.isRoot ? this.package.name : this.name,
+      version: this.package.version,
+    }
+    if (this.errors.length || !this.package.name && !this.package.version) {
+      why.errors = this.errors.length ? this.errors : [
+        new Error('invalid package: lacks name and/or version')
+      ]
```

Comment:
was this supposed to be `this.errors.length || !this.package.name || !this.package.version` ?

otherwise the error message and be just `invalid package: lacks name and version`

Reply:
Hmm, I think you're right.  It was changed from `!(this.package.name && this.package.version)`, and I did not take heed of the good Mr. DeMorgan.